lib/scheduler/thread_pool.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+module Scheduler
+  # ThreadPool manages a pool of worker threads that process tasks from a queue.
+  # It maintains a minimum number of threads and can scale up to a maximum number
+  # when there's more work to be done.
```

Comment:
I think it'll be useful to provide examples of how this thread pool is meant to be used without having to read through the implementation details.

Reply:
will do